docs/blog/2023-11-20-life-of-a-sample-part-1.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Life of a sample in Thanos, and how to configure it – Ingestion – Part I
+date: "2024-09-16"
+author: Thibault Mangé (https://github.com/thibaultmg)
+---
+
+## Life of a Sample in Thanos, and How to Configure it – Ingestion – Part I
+
+### Introduction
+
+Thanos is a sophisticated distributed system with a broad range of capabilities, and with that comes a certain level of configuration complexity. In this series of articles, we will take a deep dive into the lifecycle of a sample within Thanos, tracking its journey from initial ingestion to final retrieval. Our focus will be to explain Thanos's critical internal mechanisms and highlight the essential configurations for each component, guiding you toward achieving the operational results you are aiming for. We will be covering the following Thanos components:
+
+* **Receive**: Ingests samples from Prometheus servers and prepares them for object storage.
+* **Compactor**: Merges and deduplicates blocks in object storage.
+* **Store**: Exposes blocks in object storage for querying.
+* **Query**: Retrieves data from stores and processes queries.
+* **Query Frontend**: Distributes queries to query instances.
+
+The objective of this series of articles is to make Thanos more accessible to new users, helping to alleviate any initial apprehensions. We will also assume that the working environment is Kubernetes. Given the extensive ground to cover, our goal is to remain concise throughout this exploration.
+
+Before diving deeper, please check annexes to clarify some essential terminology. If you are already familiar with these concepts, feel free to skip ahead.
+
+### The Sample Origin: Do You Have Close Integration Capabilities?
+
+The sample usually originates from a Prometheus instance that is scraping targets in a cluster. There are two possible scenarios:
+
+* The **Prometheus instances are under your control and you can access it from your Thanos deployment**. In this case you can use the Thanos sidecar that you will attach to the pod running the prometheus server. The Thanos sidecar will directly read the raw samples from the Prometheus server using the [remote read API](https://prometheus.io/docs/prometheus/latest/querying/remote_read_api/). Then, the sidecar will behave similarly to the other scenario. It will expose its local data via the Store API as a **Receiver**, without the routing and ingestion parts. Thus, we will not go further into this use case.
```

Comment:
```suggestion
* The **Prometheus instances are under your control and you can access it from your Thanos deployment**. In this case, you can use the Thanos sidecar, which you will attach to the pod running the Prometheus server. The Thanos sidecar will directly read the raw samples from the Prometheus server using the [remote read API](https://prometheus.io/docs/prometheus/latest/querying/remote_read_api/). Then, the sidecar will behave similarly to the other scenario. It will expose its local data via the Store API as a **Receiver**, without the routing and ingestion parts. Thus, we will not delve further into this use case
```